cli/cmd/version.go

```diff
@@ -0,0 +1,103 @@
+// Package cmd provides the command-line interface for the application,
+// including the root command and its associated subcommands.
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"runtime"
+	"runtime/debug"
+
+	"github.com/spf13/cobra"
+)
+
+// BuildInfo holds information about the build, including the version,
+// commit hash, and the time when the build was created.
+type BuildInfo struct {
+	Version    string
+	Commit     string
+	CommitTime string
+	GoVersion  string
+	Goos       string
+	Goarch     string
+	Modified   bool
+}
+
+var buildInfo BuildInfo
+
+// VersionCommand represents the version command
+func VersionCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Show version and build information",
+		Long:  ``,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) > 0 {
+				fmt.Print("command not found\n\n")
+
+				err := cmd.Help()
+				if err != nil {
+					log.Printf("Failed to display help: %v", err)
+				}
+			} else {
+
+				buildInfo := CreateBuildInfo()
+
+				_, err := fmt.Fprintf(cmd.OutOrStdout(), "%s\n", buildInfo)
+				if err != nil {
+					log.Printf("Failed to write Build Information: %v", err)
+				}
+
+			}
+		},
+	}
+
+	return cmd
+}
```

Comment:
You don't need all that. Cobra has `cobra.NoArgs` field which ensures that the command does not accept any arguments. If arguments are provided, Cobra will automatically print an error message and the command's help text.

```go
// VersionCommand represents the version command
func VersionCommand() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "version",
		Short: "Show version and build information",
		Long:  "Displays detailed information about the current build of the application.",
		Args:  cobra.NoArgs, // No arguments are allowed
		Run: func(cmd *cobra.Command, args []string) {
			// Print the build information
			fmt.Println(CreateBuildInfo().String())
		},
	}

	return cmd
}
```

so it works like this:

```
$ ./retroblast version
retroblast version: "dev", commit: "84e3bf6c734f37002f8b8351783fd6f91be05986", go version: "go1.22.4", GOOS: "darwin", GOARCH: "arm64" 

$ ./retroblast version bleepblop  
Error: unknown command "bleepblop" for "retroblast version"
Usage:
  retroblast version [flags]

Flags:
  -h, --help   help for version
```